subsys/bluetooth/audio/vcs_internal.h

```diff
@@ -36,18 +36,65 @@ struct vcs_control_vol {
 	uint8_t volume;
 } __packed;
 
-int bt_vcs_client_included_get(struct bt_conn *conn,
+struct bt_vcs_client {
+	struct vcs_state state;
+	uint8_t flags;
+
+	uint16_t start_handle;
+	uint16_t end_handle;
+	uint16_t state_handle;
+	uint16_t control_handle;
+	uint16_t flag_handle;
+	struct bt_gatt_subscribe_params state_sub_params;
+	struct bt_gatt_subscribe_params flag_sub_params;
+	bool cp_retried;
+
+	bool busy;
+	struct vcs_control_vol cp_val;
+	struct bt_gatt_write_params write_params;
+	struct bt_gatt_read_params read_params;
+	struct bt_gatt_discover_params discover_params;
+	struct bt_uuid_16 uuid;
+	struct bt_conn *conn;
+
+	uint8_t vocs_inst_cnt;
+	struct bt_vocs *vocs[CONFIG_BT_VCS_CLIENT_MAX_VOCS_INST];
+	uint8_t aics_inst_cnt;
+	struct bt_aics *aics[CONFIG_BT_VCS_CLIENT_MAX_AICS_INST];
+};
+
+struct bt_vcs_server {
+	struct vcs_state state;
+	uint8_t flags;
+	struct bt_vcs_cb *cb;
+	uint8_t volume_step;
+
+	struct bt_gatt_service *service_p;
+	struct bt_vocs *vocs_insts[CONFIG_BT_VCS_VOCS_INSTANCE_COUNT];
+	struct bt_aics *aics_insts[CONFIG_BT_VCS_AICS_INSTANCE_COUNT];
+};
+
+/* Struct used as a common type for the api */
+struct bt_vcs {
+	bool client_instance;
+	union {
```

Comment:
Same as MICS, storage of the instances depend on the largest one, which might be wasteful.

Reply:
@joerchan Yes, I would like to look into this, as there are definitely some optimization possible here. 

However, can I ask to do this in a new separate PR? We have some people who would like to start using this changed API, and I would prefer not to delay an API change based on an optimization.

Reply:
That's fine with me, just wanted to point it out.

Reply:
Great :) I'll see what I can do in terms of optimizing this. If we go the bt_uuid road, I hope I can find a way to use these pointers without causing Coverity issues for each use... :D